components/renku_data_services/users/db.py

```diff
@@ -411,7 +411,9 @@ async def add_pinned_project(self, requested_by: APIUser, project_slug: str) ->
 
             if user_preferences is None:
                 new_preferences = UserPreferences(
-                    user_id=cast(str, requested_by.id), pinned_projects=PinnedProjects(project_slugs=[project_slug])
+                    user_id=cast(str, requested_by.id),
+                    pinned_projects=PinnedProjects(project_slugs=[project_slug]),
+                    dismiss_project_migration_banner=False,
```

Comment:
I think this can just be a default on the class itself so it doesn't need to be set here. Just set it `dismiss_project_migration_banner: bool = False` on the `UserPreferences` class